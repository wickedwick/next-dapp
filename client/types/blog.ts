export type BlogPost = {
  id: string
  content: string
  date: string
  title: string
}

export type BlogPostMetadata = {
  id: string
  date: string
  title: string
}

export type NearBlogPost = {
  isPublic: boolean
  name: string
  slug: string
  type: NearContentType
  values: NearField[]
  createdAt: string
  updatedAt: string
}

export type NearContentType = {
  name: string
}

export type NearField = {
  fieldType: string
  name: string
  value: string
}

export type ParamsPath = {
  id: string
}

export type PostPath = {
  params: ParamsPath
}

export type StrapiBlogPost = {
  id: number
  body: string
  postedDate?: string
  slug: string
  subtitle?: string
  title: string
  created_at: string
  updated_at: string
}
