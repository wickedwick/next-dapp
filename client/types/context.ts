import Web3 from "web3"

export type Web3ContextParams = {
  web3: Web3
  accounts: Array<string>
  contract: any
  setAccounts: (accounts: Array<string>) => void
}

export type AppContextParams = {
  isLoading: boolean
  errorMessage: string
  setLoading: (isLoading: boolean) => void
  setErrorMessage: (errorMessage: string) => void
}
