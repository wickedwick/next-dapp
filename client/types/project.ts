export type ProjectPageContent = {
  title: string
  description: string
  links: string
  documentation?: string
}
