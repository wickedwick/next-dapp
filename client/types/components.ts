import { BlogPost } from "./blog"

export type LayoutProps = {
  children: React.ReactNode
  home: boolean
}

export type PostsProps = {
  posts: BlogPost[]
}
