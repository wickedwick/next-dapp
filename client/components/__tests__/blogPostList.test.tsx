import { configure, mount } from 'enzyme'
import React from 'react'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { StrapiBlogPost } from '../../types/blog'
import BlogPostList from '../blogPostList'

configure({ adapter: new Adapter() })

describe('<BlogPostList />', () => {
  const handleShowBlog: (id: number) => void = jest.fn()
  const date = new Date(2022, 12, 31)
  const blogs: StrapiBlogPost[] = [
    {
      id: 1,
      title: 'Test Blog',
      slug: 'test-blog',
      body: 'This is a post',
      created_at: date,
      updated_at: date
    },
    {
      id: 1,
      title: 'Test Blog2',
      subtitle: 'Blog subtitle',
      postedDate: date,
      slug: 'test-blog-2',
      body: 'This is a post',
      created_at: date,
      updated_at: date
    }
  ]

  it('Renders the list of blog posts', () => {
    const wrapper = mount(<BlogPostList blogPosts={blogs} handleShowBlog={handleShowBlog} />)
    expect(wrapper.find('li').length).toBe(2)
    expect(wrapper).toMatchSnapshot()
  })

  it('Handles blog selection', () => {
    const wrapper = mount(<BlogPostList blogPosts={blogs} handleShowBlog={handleShowBlog} />)
    wrapper.find('button').first().simulate('click')
    expect(handleShowBlog).toHaveBeenCalled()
  })
})
