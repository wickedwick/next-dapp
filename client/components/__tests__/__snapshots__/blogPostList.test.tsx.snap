// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogPostList /> Renders the list of blog posts 1`] = `ReactWrapper {}`;
